src/sparse.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+#ifndef INCLUDE_sparse_h__
+#define INCLUDE_sparse_h__
+
+#include "common.h"
+
+#include "repository.h"
+#include "attr_file.h"
+
+#define GIT_SPARSE_CHECKOUT_FILE "sparse-checkout"
+typedef struct {
+	git_repository *repo;
+	git_attr_file *sparse;
+	int ignore_case;
+} git_sparse;
+
+enum {
+	GIT_SPARSE_UNCHECKED = -2,
+	GIT_SPARSE_NOTFOUND = -1,
+	GIT_SPARSE_NOCHECKOUT = 0,
+	GIT_SPARSE_CHECKOUT = 1,
+};
```

Comment:
Could we give this enum a typedef so that the proper type can be used instead of int?

Reply:
I've called it `git_sparse_status` for now, though I'm open to some better name suggestions.